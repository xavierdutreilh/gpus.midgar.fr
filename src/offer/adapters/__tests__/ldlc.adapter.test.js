jest.mock("node-fetch")

const fetch = require("node-fetch")
const fs = require("fs").promises
const { Response } = jest.requireActual("node-fetch")

const ldlc = require("../ldlc.adapter")

describe("getOffers", () => {
  beforeAll(async () => {
    fetch.mockResolvedValue(
      new Response(
        await fs.readFile(`${__dirname}/mocks/get-ldlc-products.html`, "utf8")
      )
    )
  })

  it("retrieves ldlc offers", async () => {
    const products = await ldlc.getOffers("rtx3090")
    expect(fetch).toHaveBeenCalledTimes(1)
    expect(fetch).toHaveBeenCalledWith(
      "https://www.ldlc.com/recherche/+fcat-4684+fv121-19185.html"
    )
    expect(products).toMatchSnapshot()
  })
})
