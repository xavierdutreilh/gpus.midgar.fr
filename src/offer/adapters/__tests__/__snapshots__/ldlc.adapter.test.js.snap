// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOffers retrieves ldlc offers 1`] = `
Array [
  Object {
    "name": "ASUS GeForce TUF RTX 3090 24G GAMING",
    "price": "€1,849.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "MSI GeForce RTX 3090 VENTUS 3X 24G OC",
    "price": "€1,729.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "ASUS GeForce ROG STRIX RTX 3090 O24G GAMING",
    "price": "€1,949.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "EVGA GeForce RTX 3090 XC3 BLACK GAMING",
    "price": "€1,739.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "EVGA GeForce RTX 3090 XC3 GAMING",
    "price": "€1,829.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "EVGA GeForce RTX 3090 FTW3 GAMING",
    "price": "€1,999.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "EVGA GeForce RTX 3090 FTW3 ULTRA GAMING",
    "price": "€1,999.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "MSI GeForce RTX 3090 GAMING X TRIO 24G",
    "price": "€1,849.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "Gigabyte GeForce RTX 3090 EAGLE OC 24G",
    "price": "€1,699.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "Palit GeForce RTX 3090 GamingPro",
    "price": "€1,779.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "INNO3D GeForce RTX 3090 ICHILL X4 RGB",
    "price": "€1,749.95",
    "status": "+ de 15 jours",
    "store": "ldlc",
  },
  Object {
    "name": "INNO3D GeForce RTX 3090 GAMING X3",
    "price": "€1,649.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "ASUS GeForce TUF RTX 3090 O24G GAMING",
    "price": "€1,849.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "KFA2 GeForce RTX 3090 SG (1-Click OC)",
    "price": "€1,749.95",
    "status": "+ de 15 jours",
    "store": "ldlc",
  },
  Object {
    "name": "Gigabyte GeForce RTX 3090 GAMING OC 24G",
    "price": "€1,749.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "EVGA GeForce RTX 3090 XC3 ULTRA GAMING",
    "price": "€1,899.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "ZOTAC GeForce RTX 3090 TRINITY",
    "price": "€1,729.94",
    "status": "+ de 15 jours",
    "store": "ldlc",
  },
  Object {
    "name": "INNO3D GeForce RTX 3090 ICHILL X3 RGB",
    "price": "€1,699.94",
    "status": "+ de 15 jours",
    "store": "ldlc",
  },
  Object {
    "name": "Palit GeForce RTX 3090 GamingPro OC",
    "price": "€1,789.95",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "Gainward GeForce RTX 3090 Phoenix GS (Golden Sample)",
    "price": "€1,819.94",
    "status": "Rupture",
    "store": "ldlc",
  },
  Object {
    "name": "Gainward GeForce RTX 3090 Phoenix",
    "price": "€1,729.94",
    "status": "Rupture",
    "store": "ldlc",
  },
]
`;
